finetrainers/ltx_video/ltx_video_lora.py

```diff
@@ -114,19 +127,55 @@ def prepare_latents(
     device: Optional[torch.device] = None,
     dtype: Optional[torch.dtype] = None,
     generator: Optional[torch.Generator] = None,
+    precompute: bool = False,
 ) -> torch.Tensor:
     device = device or vae.device
-    dtype = dtype or vae.dtype
 
     if image_or_video.ndim == 4:
         image_or_video = image_or_video.unsqueeze(2)
     assert image_or_video.ndim == 5, f"Expected 5D tensor, got {image_or_video.ndim}D tensor"
 
-    image_or_video = image_or_video.to(device=device, dtype=dtype)
+    image_or_video = image_or_video.to(device=device, dtype=vae.dtype)
     image_or_video = image_or_video.permute(0, 2, 1, 3, 4).contiguous()  # [B, C, F, H, W] -> [B, F, C, H, W]
-    latents = vae.encode(image_or_video).latent_dist.sample(generator=generator)
-    _, _, num_frames, height, width = latents.shape
-    latents = _normalize_latents(latents, vae.latents_mean, vae.latents_std)
+    if not precompute:
+        latents = vae.encode(image_or_video).latent_dist.sample(generator=generator)
+        latents = latents.to(dtype=dtype)
+        _, _, num_frames, height, width = latents.shape
+        latents = _normalize_latents(latents, vae.latents_mean, vae.latents_std)
```

Comment:
```suggestion
        latents = _normalize_latents(latents, vae.config.latents_mean, vae.configlatents_std)
```

Reply:
It is not part of the config. All the LTX checkpoints have it in their state dict. We cannot apply this. Especially because we are considering only hosting the latest revision of the models as diffusers-format. For all other versions, we will use single file loading.

`latents_mean` and `latents_std` are registered buffers in the model implementation.

https://github.com/huggingface/diffusers/blob/ea1ba0ba53bdd6569547e26e518f094745ed9d03/src/diffusers/models/autoencoders/autoencoder_kl_ltx.py#L806

Reply:
Oh :/